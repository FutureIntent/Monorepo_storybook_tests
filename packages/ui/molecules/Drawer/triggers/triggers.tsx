import {
    AnimationInterface,
    CloseHandlerInterface,
    DragEndInterface,
    OpenHandlerInterface,
} from '../types/drawerTypes';

const transition = {
    bounce: 0,
};

const handleOpen = ({ axis, direction, setOpen, controls }: OpenHandlerInterface) => {
    const innerLength = axis === 'x' ? window.innerWidth : window.innerHeight;

    controls
        .start({
            [axis]: (innerLength / 2) * direction,
            transition,
        })
        .then(() => setOpen(true));
};

const handleClose = ({ axis, setOpen, controls }: CloseHandlerInterface) => {
    controls
        .start({
            [axis]: 0,
            transition,
        })
        .then(() => setOpen(false));
};

export const handleDragEnd = ({
    info,
    pullerHeight,
    side,
    controls,
    setOpen,
}: DragEndInterface) => {
    const axis: string = ['right', 'left'].includes(side) ? 'x' : 'y';
    const direction: number = ['top', 'left'].includes(side) ? 1 : -1;
    let initialPoint = 0;
    const pullerMiddle =
        axis === 'x'
            ? (window.innerWidth / 2 + pullerHeight) / 2
            : (window.innerHeight / 2 + pullerHeight) / 2;

    if (direction === -1 && axis === 'x') {
        initialPoint = window.innerWidth;
    } else if (direction === -1 && axis === 'y') {
        initialPoint = window.innerHeight;
    }

    const splitPointModule = Math.abs(initialPoint - pullerMiddle);

    if (axis === 'x') {
        if (direction === 1 && info.point.x >= splitPointModule) {
            handleOpen({ axis, direction, setOpen, controls });
        } else if (direction === 1 && info.point.x <= splitPointModule) {
            handleClose({ axis, setOpen, controls });
        }

        if (direction === -1 && info.point.x <= splitPointModule) {
            handleOpen({ axis, direction, setOpen, controls });
        } else if (direction === -1 && info.point.x >= splitPointModule) {
            handleClose({ axis, setOpen, controls });
        }
    }

    if (axis === 'y') {
        if (direction === 1 && info.point.y >= splitPointModule) {
            handleOpen({ axis, direction, setOpen, controls });
        } else if (direction === 1 && info.point.y <= splitPointModule) {
            handleClose({ axis, setOpen, controls });
        }

        if (direction === -1 && info.point.y <= splitPointModule) {
            handleOpen({ axis, direction, setOpen, controls });
        } else if (direction === -1 && info.point.y >= splitPointModule) {
            handleClose({ axis, setOpen, controls });
        }
    }
};

export const handleAnimation = ({ side, open, setOpen, controls }: AnimationInterface) => {
    const direction: number = ['top', 'left'].includes(side) ? 1 : -1;
    const axis: string = ['right', 'left'].includes(side) ? 'x' : 'y';

    if (!open) {
        handleOpen({ axis, direction, setOpen, controls });
    } else {
        handleClose({ axis, setOpen, controls });
    }
};
