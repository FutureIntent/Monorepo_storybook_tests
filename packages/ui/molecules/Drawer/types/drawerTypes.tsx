import { AnimationControls, DragControls, PanInfo } from 'framer-motion';
import React from 'react';

export type SideType = 'top' | 'right' | 'bottom' | 'left';

export interface DrawerInterface {
    side?: SideType;
    pullerHeight?: number;
    pullerColor?: string;
    pullerRadius?: number;
    lineLength?: number;
    lineColor?: string;
    children?: JSX.Element;
    isOpen?: boolean;
}

export interface PullerWrapperInterface {
    pullerHeight: number;
    pullerColor: string;
    pullerRadius: number;
}

export interface PullerLineInterface {
    isAxisX: boolean;
    lineColor: string;
    lineLength: number;
}

export interface PullerInterface {
    pullerHeight: number;
    pullerColor: string;
    pullerRadius: number;
    lineLength: number;
    lineColor: string;
    side: SideType;
    dragControls: DragControls;
}

export interface ContentInterface {
    children?: JSX.Element;
    side: SideType;
}
export interface DragConstraintsInterface {
    side: SideType;
}

export interface InitialPositionInterface {
    side: SideType;
}

export interface DragEndInterface {
    info: PanInfo;
    pullerHeight: number;
    side: SideType;
    controls: AnimationControls;
    setOpen: React.Dispatch<React.SetStateAction<boolean>>;
}

export interface AnimationInterface {
    side: SideType;
    open: boolean;
    setOpen: React.Dispatch<React.SetStateAction<boolean>>;
    controls: AnimationControls;
}

export interface OpenHandlerInterface {
    axis: string;
    direction: number;
    setOpen: React.Dispatch<React.SetStateAction<boolean>>;
    controls: AnimationControls;
}

export interface CloseHandlerInterface {
    axis: string;
    setOpen: React.Dispatch<React.SetStateAction<boolean>>;
    controls: AnimationControls;
}
