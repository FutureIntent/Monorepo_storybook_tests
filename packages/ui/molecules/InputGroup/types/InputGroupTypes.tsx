export interface InputGroupInterface extends React.HTMLAttributes<HTMLDivElement> {
    children: JSX.Element[];
    childrenLength?: string[];
}

export interface DateGroupInterface {
    month: string;
    year: string;
}

export interface PasswordGroupInterface {
    password: string;
    repeat: string;
}

export interface MultiVariationInterface {
    input1: string;
    input2: string;
    input3: string;
}

export interface GroupContainerInterface {
    repeat: number;
    split: string;
}

export interface PhoneNumberInterface {
    prefix: string;
    phone: string;
}
