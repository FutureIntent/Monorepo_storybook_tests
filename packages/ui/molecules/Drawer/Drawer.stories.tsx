import { ComponentMeta, ComponentStory } from '@storybook/react';
import { useEffect } from 'react';
import { useResize } from '../../hooks/useResize';
import Typography from '../../atoms/Typography/Typography';
import Drawer from './Drawer';
import { DrawerInterface } from './types/drawerTypes';

export default {
    title: 'Drawer',
    component: Drawer,
    argTypes: {
        side: {
            name: 'Drawer side',
            options: ['top', 'right', 'bottom', 'left'],
            control: {
                type: 'radio',
            },
            defaultValue: 'bottom',
        },
        pullerHeight: {
            name: 'Puller height',
            control: {
                type: 'number',
            },
            defaultValue: 30,
        },
        pullerColor: {
            name: 'Puller color',
            control: {
                type: 'color',
            },
            defaultValue: 'rgba(255, 0, 0, 0.8)',
        },
        pullerRadius: {
            name: 'Puller radius',
            control: {
                type: 'number',
            },
            defaultValue: 20,
        },
        lineLength: {
            name: 'Line length',
            control: {
                type: 'number',
            },
            defaultValue: 60,
        },
        lineColor: {
            name: 'Line color',
            control: {
                type: 'color',
            },
            defaultValue: 'rgba(242, 255, 0, 1)',
        },
        isOpen: {
            name: 'Is open',
            control: {
                type: 'boolean',
            },
            defaultValue: false,
        },
        child: {
            table: {
                disable: true,
            },
        },
    },
} as ComponentMeta<typeof Drawer>;

export const Basic: ComponentStory<typeof Drawer> = (props: DrawerInterface) => {
    const { width, height } = useResize();

    useEffect(() => {
        const root: HTMLElement | null = document.getElementById('root');

        if (root) root.style.padding = '0';

        return () => {
            if (root) root.style.padding = '1rem';
        };
    }, []);

    return (
        <div style={{ width: `${width}px`, height: `${height}px` }}>
            <Drawer {...props}>
                <div
                    style={{
                        width: '100%',
                        height: '100%',
                        backgroundColor: 'rgba(0, 0, 0, 0.7)',
                        backdropFilter: 'blur( 10px )',
                        display: 'flex',
                        justifyContent: 'center',
                        alignItems: 'center',
                    }}
                >
                    <Typography color="white">CONTENT</Typography>
                </div>
            </Drawer>
        </div>
    );
};
