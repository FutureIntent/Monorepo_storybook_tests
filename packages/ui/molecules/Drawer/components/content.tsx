import styled from 'styled-components';
import { ContentInterface } from '../types/drawerTypes';

const Wrapper = styled.div.attrs((props: any) => ({
    side: props.side,
}))`
    overflow: auto;
    user-select: none;
    ${({ side }) =>
        side === 'top' || side === 'bottom'
            ? 'width: 100vw; height: 50vh'
            : 'width: 50vw; height: 100vh'};
`;

const ContentContainer = ({ children, side }: ContentInterface) => (
    <Wrapper side={side} data-testid="contentContainer">
        {children}
    </Wrapper>
);

export default ContentContainer;
