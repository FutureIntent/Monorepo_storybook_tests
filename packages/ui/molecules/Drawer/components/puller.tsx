import styled from 'styled-components';
import { PullerInterface, PullerLineInterface, PullerWrapperInterface } from '../types/drawerTypes';

const Wrapper = styled.div<PullerWrapperInterface>`
    align-items: center;
    background-color: ${({ pullerColor }) => pullerColor};
    display: flex;
    justify-content: center;

    &.top {
        border-radius: ${({ pullerRadius }) => `0 0 ${pullerRadius}px ${pullerRadius}px`};
        height: ${({ pullerHeight }) => pullerHeight}px;
    }

    &.right {
        border-radius: ${({ pullerRadius }) => `${pullerRadius}px 0 0 ${pullerRadius}px`};
        width: ${({ pullerHeight }) => pullerHeight}px;
    }

    &.bottom {
        border-radius: ${({ pullerRadius }) => `${pullerRadius}px ${pullerRadius}px 0 0`};
        height: ${({ pullerHeight }) => pullerHeight}px;
    }

    &.left {
        border-radius: ${({ pullerRadius }) => `0 ${pullerRadius}px ${pullerRadius}px 0`};
        width: ${({ pullerHeight }) => pullerHeight}px;
    }
`;

const Line = styled.hr<PullerLineInterface>`
    border-color: ${({ lineColor }) => lineColor};

    ${({ isAxisX, lineLength }) =>
        isAxisX
            ? `
    width: 1%;
    height: ${lineLength}px;
    `
            : `
    width: ${lineLength}px;
    height: 1%;
    `}
`;

const PullerContainer = ({
    pullerHeight,
    pullerColor,
    pullerRadius,
    lineLength,
    lineColor,
    side,
    dragControls,
}: PullerInterface) => {
    const startDrag = (event: React.PointerEvent<HTMLDivElement>) => {
        dragControls.start(event, { snapToCursor: false });
    };

    return (
        <Wrapper
            data-testid="pullerContainer"
            pullerHeight={pullerHeight}
            pullerColor={pullerColor}
            pullerRadius={pullerRadius}
            className={side}
            onPointerDown={startDrag}
        >
            <Line
                isAxisX={!!(side === 'right' || side === 'left')}
                lineLength={lineLength}
                lineColor={lineColor}
                data-testid="lineContainer"
            />
        </Wrapper>
    );
};

export default PullerContainer;
