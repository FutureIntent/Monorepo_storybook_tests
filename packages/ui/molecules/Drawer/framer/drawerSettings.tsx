import { SideType, DragConstraintsInterface, InitialPositionInterface } from '../types/drawerTypes';

export const dragConstraints = ({ side }: DragConstraintsInterface) => {
    if (side === 'top') return { top: 0, bottom: window.innerHeight / 2 };
    if (side === 'right') return { left: -window.innerWidth / 2, right: 0 };
    if (side === 'bottom') return { top: -window.innerHeight / 2, bottom: 0 };
    if (side === 'left') return { left: 0, right: window.innerWidth / 2 };

    return {
        top: 0,
        right: 0,
        bottom: 0,
        left: 0,
    };
};

export const dragElastic = (side: SideType) => ({ [side]: 0.5 });

export const initialPosition = ({ side }: InitialPositionInterface) => {
    if (side === 'top') return { y: window.innerHeight / 2 };
    if (side === 'right') return { x: -window.innerWidth / 2 };
    if (side === 'bottom') return { y: -window.innerHeight / 2 };
    if (side === 'left') return { x: window.innerWidth / 2 };

    return false;
};
