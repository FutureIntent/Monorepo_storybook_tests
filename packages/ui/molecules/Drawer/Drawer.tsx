import { useEffect, useState } from 'react';
import styled from 'styled-components';
import { motion, useAnimationControls, useDragControls } from 'framer-motion';
import shortid from 'shortid';
import { DrawerInterface } from './types/drawerTypes';
import ContentContainer from './components/content';
import PullerContainer from './components/puller';
import { dragConstraints, dragElastic, initialPosition } from './framer/drawerSettings';
import { useResize } from '../../hooks/useResize';
import { handleAnimation, handleDragEnd } from './triggers/triggers';

const DrawerContainer = styled.div`
    display: flex;
    position: fixed;
    touch-action: none;
    user-select: none;

    &.top {
        flex-direction: column-reverse;
        top: -50vh;
    }

    &.right {
        flex-direction: row;
        right: -50vw;
    }

    &.bottom {
        bottom: -50vh;
        flex-direction: column;
    }

    &.left {
        flex-direction: row-reverse;
        left: -50vw;
    }
`;

const Drawer = ({
    side = 'bottom',
    pullerHeight = 30,
    pullerColor = 'rgba(255, 0, 0, 0.8)',
    lineLength = 60,
    pullerRadius = 20,
    lineColor = 'rgba(242, 255, 0, 1)',
    isOpen = false,
    children,
}: DrawerInterface) => {
    const dragControls = useDragControls();
    const controls = useAnimationControls();

    const [open, setOpen] = useState<boolean>(isOpen);

    // Motion drawer's component rerender on resize in case no external listener is used
    useResize();

    useEffect(() => {
        if (isOpen !== open) handleAnimation({ side, open, setOpen, controls });
    }, [isOpen]);

    return (
        <DrawerContainer
            as={motion.div}
            initial={open ? initialPosition({ side }) : false}
            animate={controls}
            drag={side === 'top' || side === 'bottom' ? 'y' : 'x'}
            dragControls={dragControls}
            dragListener={false}
            dragConstraints={dragConstraints({ side })}
            dragElastic={dragElastic(side)}
            dragMomentum={false}
            onDragEnd={(_, info) => handleDragEnd({ info, pullerHeight, side, controls, setOpen })}
            className={side}
            // Neither the most elegant nor optimised solution, but on resize motion components lose drag constraints and ignore prop overrides. Required a full component rerender.
            // I didn't find other workarounds.
            // https://github.com/framer/motion/issues/1659
            key={shortid.generate()}
            data-testid="drawerContainer"
        >
            <PullerContainer
                pullerHeight={pullerHeight}
                pullerColor={pullerColor}
                pullerRadius={pullerRadius}
                lineLength={lineLength}
                lineColor={lineColor}
                side={side}
                dragControls={dragControls}
            />
            <ContentContainer side={side}>{children}</ContentContainer>
        </DrawerContainer>
    );
};

export default Drawer;
