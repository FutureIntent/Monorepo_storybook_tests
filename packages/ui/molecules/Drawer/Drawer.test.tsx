import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Drawer from './Drawer';
import { handleAnimation, handleDragEnd } from './triggers/triggers';
import { DrawerInterface } from './types/drawerTypes';
import * as Settings from './framer/drawerSettings';
import * as Puller from './components/puller';
import * as Content from './components/content';

const defaultProps: DrawerInterface = {
    side: 'top',
    pullerHeight: 30,
    pullerColor: 'rgba(255, 0, 0, 0.8)',
    pullerRadius: 20,
    lineLength: 60,
    lineColor: 'rgba(242, 255, 0, 1)',
    isOpen: true,
};

jest.mock('./triggers/triggers', () => ({
    __esModule: true,
    handleAnimation: jest.fn(),
    handleDragEnd: jest.fn(),
}));

describe('Drawer is flawlessly functional if', () => {
    it('renders and provides appropriate props to children', () => {
        const puller = jest.spyOn(Puller, 'default');
        const content = jest.spyOn(Content, 'default');

        render(<Drawer {...defaultProps} />);

        // empty object is some sort of React's dark magic under the hood
        expect(puller).toHaveBeenCalledWith(
            {
                pullerHeight: 30,
                pullerColor: 'rgba(255, 0, 0, 0.8)',
                pullerRadius: 20,
                lineLength: 60,
                lineColor: 'rgba(242, 255, 0, 1)',
                side: 'top',
                dragControls: expect.any(Object),
            },
            {},
        );

        // empty object is some sort of React's dark magic under the hood
        expect(content).toHaveBeenCalledWith(
            {
                children: undefined,
                side: 'top',
            },
            {},
        );

        const drawerContainer = screen.getByTestId('drawerContainer');
        const pullerContainer = screen.getByTestId('pullerContainer');
        const contentContainer = screen.getByTestId('contentContainer');

        expect(drawerContainer).toBeInTheDocument();
        expect(pullerContainer).toBeInTheDocument();
        expect(contentContainer).toBeInTheDocument();
    });

    it('changes styles accordingly to props', () => {
        const drawerOpened = renderer.create(<Drawer {...defaultProps} isOpen />);

        const drawerOpenedTree = drawerOpened.toJSON();
        expect(drawerOpenedTree).toMatchSnapshot();

        const { rerender } = render(<Drawer {...defaultProps} isOpen={false} />);

        let container = screen.getByTestId('drawerContainer');
        let puller = screen.getByTestId('pullerContainer');
        let pullerLine = screen.getByTestId('lineContainer');
        let content = screen.getByTestId('contentContainer');

        expect(getComputedStyle(container).flexDirection).toEqual('column-reverse');
        expect(getComputedStyle(container).top).toEqual('-50vh');

        expect(getComputedStyle(puller).height).toEqual('30px');
        expect(getComputedStyle(puller).backgroundColor).toEqual(defaultProps.pullerColor);
        expect(getComputedStyle(puller).borderRadius).toEqual('0 0 20px 20px');

        expect(getComputedStyle(pullerLine).width).toEqual('60px');
        expect(getComputedStyle(pullerLine).height).toEqual('1%');
        expect(getComputedStyle(pullerLine).borderColor).toEqual(
            defaultProps.lineColor?.replaceAll(/\s{1,}/g, ''),
        );

        expect(getComputedStyle(content).width).toEqual('100vw');
        expect(getComputedStyle(content).height).toEqual('50vh');

        rerender(<Drawer {...defaultProps} isOpen={false} side="right" />);

        container = screen.getByTestId('drawerContainer');
        puller = screen.getByTestId('pullerContainer');
        pullerLine = screen.getByTestId('lineContainer');
        content = screen.getByTestId('contentContainer');

        expect(getComputedStyle(container).flexDirection).toEqual('row');
        expect(getComputedStyle(container).right).toEqual('-50vw');

        expect(getComputedStyle(puller).width).toEqual('30px');
        expect(getComputedStyle(puller).backgroundColor).toEqual(defaultProps.pullerColor);
        expect(getComputedStyle(puller).borderRadius).toEqual('20px 0 0 20px');

        expect(getComputedStyle(pullerLine).height).toEqual('60px');
        expect(getComputedStyle(pullerLine).width).toEqual('1%');
        expect(getComputedStyle(pullerLine).borderColor).toEqual(
            defaultProps.lineColor?.replaceAll(/\s{1,}/g, ''),
        );

        expect(getComputedStyle(content).width).toEqual('50vw');
        expect(getComputedStyle(content).height).toEqual('100vh');

        rerender(<Drawer {...defaultProps} isOpen={false} side="bottom" />);

        container = screen.getByTestId('drawerContainer');
        puller = screen.getByTestId('pullerContainer');
        pullerLine = screen.getByTestId('lineContainer');
        content = screen.getByTestId('contentContainer');

        expect(getComputedStyle(container).flexDirection).toEqual('column');
        expect(getComputedStyle(container).bottom).toEqual('-50vh');

        expect(getComputedStyle(puller).height).toEqual('30px');
        expect(getComputedStyle(puller).backgroundColor).toEqual(defaultProps.pullerColor);
        expect(getComputedStyle(puller).borderRadius).toEqual('20px 20px 0 0');

        expect(getComputedStyle(pullerLine).width).toEqual('60px');
        expect(getComputedStyle(pullerLine).height).toEqual('1%');
        expect(getComputedStyle(pullerLine).borderColor).toEqual(
            defaultProps.lineColor?.replaceAll(/\s{1,}/g, ''),
        );

        expect(getComputedStyle(content).width).toEqual('100vw');
        expect(getComputedStyle(content).height).toEqual('50vh');

        rerender(<Drawer {...defaultProps} isOpen={false} side="left" />);

        container = screen.getByTestId('drawerContainer');
        puller = screen.getByTestId('pullerContainer');
        pullerLine = screen.getByTestId('lineContainer');
        content = screen.getByTestId('contentContainer');

        expect(getComputedStyle(container).flexDirection).toEqual('row-reverse');
        expect(getComputedStyle(container).left).toEqual('-50vw');

        expect(getComputedStyle(puller).width).toEqual('30px');
        expect(getComputedStyle(puller).backgroundColor).toEqual(defaultProps.pullerColor);
        expect(getComputedStyle(puller).borderRadius).toEqual('0 20px 20px 0');

        expect(getComputedStyle(pullerLine).height).toEqual('60px');
        expect(getComputedStyle(pullerLine).width).toEqual('1%');
        expect(getComputedStyle(pullerLine).borderColor).toEqual(
            defaultProps.lineColor?.replaceAll(/\s{1,}/g, ''),
        );

        expect(getComputedStyle(content).width).toEqual('50vw');
        expect(getComputedStyle(content).height).toEqual('100vh');
    });

    it('calls triggers and functions with appropriate object attributes', () => {
        const dragConstraints = jest.spyOn(Settings, 'dragConstraints');
        const dragElastic = jest.spyOn(Settings, 'dragElastic');
        const initialPosition = jest.spyOn(Settings, 'initialPosition');

        const { rerender } = render(<Drawer {...defaultProps} />);

        expect(dragConstraints).toHaveReturnedWith({
            top: 0,
            bottom: window.innerHeight / 2,
        });

        expect(dragElastic).toHaveReturnedWith({
            top: 0.5,
        });

        expect(initialPosition).toHaveReturnedWith({
            y: window.innerHeight / 2,
        });

        rerender(<Drawer {...defaultProps} side="right" />);

        expect(dragConstraints).toHaveReturnedWith({
            left: -window.innerWidth / 2,
            right: 0,
        });

        expect(dragElastic).toHaveReturnedWith({
            right: 0.5,
        });

        expect(initialPosition).toHaveReturnedWith({
            x: -window.innerWidth / 2,
        });

        rerender(<Drawer {...defaultProps} side="bottom" />);

        expect(dragConstraints).toHaveReturnedWith({
            top: -window.innerHeight / 2,
            bottom: 0,
        });

        expect(dragElastic).toHaveReturnedWith({
            bottom: 0.5,
        });

        expect(initialPosition).toHaveReturnedWith({
            y: -window.innerHeight / 2,
        });

        rerender(<Drawer {...defaultProps} side="left" isOpen={false} />);

        expect(dragConstraints).toHaveReturnedWith({
            left: 0,
            right: window.innerWidth / 2,
        });

        expect(dragElastic).toHaveReturnedWith({
            left: 0.5,
        });

        expect(initialPosition).toHaveReturnedWith({
            x: window.innerWidth / 2,
        });

        expect(handleAnimation).toHaveBeenLastCalledWith({
            side: 'left',
            open: true,
            setOpen: expect.anything(),
            controls: expect.anything(),
        });

        const pullerContainer = screen.getByTestId('pullerContainer');

        pullerContainer.addEventListener('pointerup', () =>
            handleDragEnd({
                info: null,
                pullerHeight: 30,
                side: 'left',
                controls: null,
                setOpen: null,
            } as any),
        );

        fireEvent.pointerUp(pullerContainer);

        expect(handleDragEnd).toHaveBeenLastCalledWith({
            info: null,
            pullerHeight: 30,
            side: 'left',
            controls: null,
            setOpen: null,
        });
    });
});
