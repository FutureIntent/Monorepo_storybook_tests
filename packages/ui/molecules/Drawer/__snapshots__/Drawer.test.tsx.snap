// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer is flawlessly functional if changes styles accordingly to props 1`] = `
<div
  className="sc-hLBbgP Bttir top"
  data-projection-id={1}
  data-testid="drawerContainer"
  style={
    {
      "transform": "translateY(384px) translateZ(0)",
    }
  }
>
  <div
    className="sc-gswNZR ccrzIO top"
    data-testid="pullerContainer"
    onPointerDown={[Function]}
  >
    <hr
      className="sc-dkrFOg cyywsF"
      data-testid="lineContainer"
    />
  </div>
  <div
    className="sc-bcXHqe eYAEdL"
    data-testid="contentContainer"
  />
</div>
`;
